crates/symbolicator/src/services/cacher.rs

```diff
@@ -477,6 +462,29 @@ impl<T: CacheItemRequest> Cacher<T> {
             } else {
                 // A concurrent cache lookup is considered new. This does not imply a cache miss.
                 metric!(counter(&format!("caches.{}.channel.miss", name)) += 1);
+
+                // We count down towards zero, and if we reach or surpass it, we will short circuit here.
+                // Doing the short-circuiting here means we don't create a channel at all, and don't
+                // put it into `current_computations`.
+                // Previously, that could have potentially returned the error to a non-lazy
+                // computation requests based on the `key`.
```

Comment:
not much point leaving something like this in a comment.  better to write it in the commit message